Accessing a null MeshInstance object in BakedLightmap instead of the GeometryInstance

### Godot version

3.x

### System information

Linux

### Issue description

`mi` has been checked/tested to be null, yet is accessed on the following lines:

https://github.com/godotengine/godot/blob/40a348bce2171d2c7f73886630df5f4e8caf2daf/scene/3d/baked_lightmap.cpp#L392
https://github.com/godotengine/godot/blob/40a348bce2171d2c7f73886630df5f4e8caf2daf/scene/3d/baked_lightmap.cpp#L417
https://github.com/godotengine/godot/blob/40a348bce2171d2c7f73886630df5f4e8caf2daf/scene/3d/baked_lightmap.cpp#L418

Should these variables be referencing `gi`?

### Steps to reproduce

Compile with `-Wnonnull`. 4.0 doesn't look to be affected.

The warning is somewhat hard to trigger due to the UB involved.
Using the following command `scons -j16 platform=x11 tools=no target=release_debug CCFLAGS=-Wnonnull` after touching `scene/3d/baked_lightmap.cpp` produces (for me):

```cpp
scene/3d/baked_lightmap.cpp: In member function 'void BakedLightmap::_find_meshes_and_lights(Node*, Vector<BakedLightmap::MeshesFound>&, Vector<BakedLightmap::LightsFound>&)':
scene/3d/baked_lightmap.cpp:392:73: warning: 'this' pointer is null [-Wnonnull]
  392 |                                 all_override = mi->get_material_override();
      |                                                ~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from scene/3d/multimesh_instance.h:34,
                 from scene/3d/baked_lightmap.h:35,
                 from scene/3d/baked_lightmap.cpp:31:
./scene/3d/visual_instance.h:151:23: note: in a call to non-static member function 'Ref<Material> GeometryInstance::get_material_override() const'
  151 |         Ref<Material> get_material_override() const;
      |                       ^~~~~~~~~~~~~~~~~~~~~
```
```cpp
scene/3d/baked_lightmap.cpp:417:87: warning: 'this' pointer is null [-Wnonnull]
  417 |                                         mf.cast_shadows = mi->get_cast_shadows_setting() != GeometryInstance::SHADOW_CASTING_SETTING_OFF;

      |                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from scene/3d/multimesh_instance.h:34,
                 from scene/3d/baked_lightmap.h:35,
                 from scene/3d/baked_lightmap.cpp:31:
./scene/3d/visual_instance.h:130:30: note: in a call to non-static member function 'GeometryInstance::ShadowCastingSetting GeometryInstance::get_cast_shadows_setting() const'
  130 |         ShadowCastingSetting get_cast_shadows_setting() const;
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~
```
```cpp
scene/3d/baked_lightmap.cpp:418:89: warning: 'this' pointer is null [-Wnonnull]
  418 |                                         mf.generate_lightmap = mi->get_generate_lightmap();
      |                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from scene/3d/multimesh_instance.h:34,
                 from scene/3d/baked_lightmap.h:35,
                 from scene/3d/baked_lightmap.cpp:31:
./scene/3d/visual_instance.h:133:14: note: in a call to non-static member function 'bool GeometryInstance::get_generate_lightmap()'
  133 |         bool get_generate_lightmap();
      |              ^~~~~~~~~~~~~~~~~~~~~
```

### Minimal reproduction project

_No response_

--- a/scene/3d/baked_lightmap.cpp
+++ b/scene/3d/baked_lightmap.cpp
@@ -389,7 +389,7 @@ void BakedLightmap::_find_meshes_and_lights(Node *p_at_node, Vector<MeshesFound>
 
 			GeometryInstance *gi = Object::cast_to<GeometryInstance>(p_at_node);
 			if (gi) {
-				all_override = mi->get_material_override();
+				all_override = gi->get_material_override();
 			}
 
 			for (int i = 0; i < bmeshes.size(); i += 2) {
@@ -414,8 +414,8 @@ void BakedLightmap::_find_meshes_and_lights(Node *p_at_node, Vector<MeshesFound>
 				mf.mesh = mesh;
 
 				if (gi) {
-					mf.cast_shadows = mi->get_cast_shadows_setting() != GeometryInstance::SHADOW_CASTING_SETTING_OFF;
-					mf.generate_lightmap = mi->get_generate_lightmap();
+					mf.cast_shadows = gi->get_cast_shadows_setting() != GeometryInstance::SHADOW_CASTING_SETTING_OFF;
+					mf.generate_lightmap = gi->get_generate_lightmap();
 				} else {
 					mf.cast_shadows = true;
 					mf.generate_lightmap = true;

